src/main/java/org/jabref/gui/mergeentries/FetchAndMergeEntry.java

```diff
@@ -70,38 +70,143 @@ public void fetchAndMerge(BibEntry entry, Field field) {
+                      .onFailure(exception -> handleFetchException(exception, fetcher, entry))
+                      .executeWith(taskExecutor);
+    }
+
+    private void processFetchedEntryWithDialog(Optional<BibEntry> fetchedEntry, Field field, String fieldContent, BibEntry originalEntry, IdBasedFetcher fetcher) {
+        ImportCleanup cleanup = ImportCleanup.targeting(bibDatabaseContext.getMode(), preferences.getFieldPreferences());
+        if (fetchedEntry.isPresent()) {
+            cleanup.doPostCleanup(fetchedEntry.get());
+            showMergeDialog(originalEntry, fetchedEntry.get(), fetcher);
+        } else {
+            dialogService.notify(Localization.lang("Cannot get info based on given %0: %1", field.getDisplayName(), fieldContent));
+        }
+    }
+
+    public void fetchAndMergeBatch(List<BibEntry> entries) {
+        entries.forEach(entry -> SUPPORTED_FIELDS.forEach(field -> fetchAndMergeEntry(entry, field)));
+    }
+
+    private void fetchAndMergeEntry(BibEntry entry, Field field) {
+        entry.getField(field).ifPresentOrElse(
+                fieldContent -> WebFetchers.getIdBasedFetcherForField(field, preferences.getImportFormatPreferences()).ifPresent(
+                        fetcher -> executeFetchTask(fetcher, field, fieldContent, entry)
+                ),
+                () -> {
+                    if (hasAnySupportedField(entry)) {
+                        dialogService.notify(Localization.lang("No %0 found", field.getDisplayName()));
+                    }
                 }
+        );
+    }
+
+    private boolean hasAnySupportedField(BibEntry entry) {
+        return SUPPORTED_FIELDS.stream().noneMatch(field -> entry.getField(field).isPresent());
+    }
+
+    private void executeFetchTask(IdBasedFetcher fetcher, Field field, String fieldContent, BibEntry entry) {
+        BackgroundTask.wrap(() -> fetcher.performSearchById(fieldContent))
+                      .onSuccess(fetchedEntry -> processFetchedEntry(fetchedEntry, field, fieldContent, entry))
+                      .onFailure(exception -> handleFetchException(exception, fetcher, entry))
+                      .executeWith(taskExecutor);
+    }
+
+    private void processFetchedEntry(Optional<BibEntry> fetchedEntry, Field field, String fieldContent, BibEntry originalEntry) {
+        if (fetchedEntry.isPresent()) {
+            ImportCleanup cleanup = ImportCleanup.targeting(bibDatabaseContext.getMode(), preferences.getFieldPreferences());
+            cleanup.doPostCleanup(fetchedEntry.get());
+            mergeWithoutDialog(originalEntry, fetchedEntry.get());
+        } else {
+            dialogService.notify(Localization.lang("Cannot get info based on given %0: %1", field.getDisplayName(), fieldContent));
+        }
+    }
+
+    private void handleFetchException(Exception exception, IdBasedFetcher fetcher, BibEntry entry) {
+        if (hasAnySupportedField(entry)) {
```

Comment:
I think, this is not required. The id fetcher should only be called if the id is present. If id present, the exception is shown.

--> remove this if (and remove `hasAnySupportedField` alltogether)